File: backend/goat_listing/src/main/java/com/goat/cart/domain/exceptions/CartNotFoundException.java
package com.goat.cart.domain.exceptions;

/**
 * Excepción lanzada cuando no se encuentra el carrito del usuario.
 */
public class CartNotFoundException extends RuntimeException {
    public CartNotFoundException(String message) {
        super(message);
    }

    public CartNotFoundException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: backend/goat_catalog/src/goat_catalog/cart/domain/exceptions/cart_not_found_exception.py
"""Excepción cuando no se encuentra un carrito."""


class CartNotFoundException(Exception):
    """Excepción lanzada cuando un carrito no existe."""

    def __init__(self, message: str = "Carrito no encontrado") -> None:
        """Inicializa la excepción.

        Args:
            message: Mensaje de error
        """
        super().__init__(message)
        self.message = message

